### DISCUSSION TITLE
vector matching always return empty vector

### DISCUSSION THREAD

**Discussion by gucki** [prometheus/prometheus#10362] [Q&A]
I try to add up the rates of various error counters. Each metric contains entries with the same disk/ instances, which I verified by running the individual rate(..) queries alone. However, trying to add them always returns in an empty vector. Thus I assume this is a bug.

rate(smartmon_reallocated_sector_ct_raw_value[5m]) +
on (disk, instance) rate(smartmon_reallocated_event_count_raw_value[5m])

Version 2.33.1

**Comment by roidelapluie** [comment]
Can you please provide us example of metrics? Thanks.
